dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/AbstractOracleConnectorTest.java

```diff
@@ -44,4 +48,68 @@ public void buildProperties_success() {
     assertEquals(EXAMPLE_PASSWORD, actual.getProperty("password"));
     assertEquals(EXAMPLE_USER, actual.getProperty("user"));
   }
+
+  @Test
+  public void buildUrl_serviceNameAndSidBothProvided_throwsException() {
+    Exception serviceOrSid = OracleConnectorExceptions.mustProvideServiceOrSid();
+    when(arguments.getOracleServicename()).thenReturn("ORCLPDB");
+    when(arguments.getOracleSID()).thenReturn("ORCLPDB1");
+
+    // Act
+    ThrowingRunnable runnable = () -> AbstractOracleConnector.buildUrl(arguments);
+    Exception exception = assertThrows(MetadataDumperUsageException.class, runnable);
+
+    // Assert
+    assertEquals(serviceOrSid.getMessage(), exception.getMessage());
+  }
+
+  @Test
+  public void buildUrl_serviceNameAndSidBothNull_throwsException() {
+    Exception serviceOrSid = OracleConnectorExceptions.mustProvideServiceOrSid();
+
+    // Act
+    ThrowingRunnable runnable = () -> AbstractOracleConnector.buildUrl(arguments);
+    Exception exception = assertThrows(MetadataDumperUsageException.class, runnable);
+
+    // Assert
+    assertEquals(serviceOrSid.getMessage(), exception.getMessage());
```

Comment:
I would suggest writing the actual message that you want to assert against, instead of using the result of the call to the method in `OracleConnectorExceptions`.

See examples in other test classes, e.g. https://github.com/google/dwh-migration-tools/blob/7399219ad87ccfd15bf16e2eae702e5ba00c4cff/dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/SplitTextColumnQueryGeneratorTest.java#L246-L249

Reply:
Done